src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -423,40 +423,95 @@ private static bool TryParseExactD(ReadOnlySpan<char> guidString, ref GuidResult
 
             ref Guid g = ref result._parsedGuid;
 
-            uint uintTmp;
-            if (TryParseHex(guidString.Slice(0, 8), out Unsafe.As<int, uint>(ref g._a)) && // _a
-                TryParseHex(guidString.Slice(9, 4), out uintTmp)) // _b
+            unsafe
             {
-                g._b = (short)uintTmp;
-
-                if (TryParseHex(guidString.Slice(14, 4), out uintTmp)) // _c
+                byte* bytes = stackalloc byte[4];
+                unchecked
                 {
-                    g._c = (short)uintTmp;
+                    var buffer = guidString;
+                    if (!Number.TryParseHexStrict(buffer[0], buffer[1], ref bytes[3]))
+                    {
+                        if (!Number.TryParseHexForTwoBytes(buffer, 0, ref bytes[3], ref bytes[2]))
+                            goto FalseExit;
+                    }
+                    else
+                    {
+                        if (!Number.TryParseHexStrict(buffer[2], buffer[3], ref bytes[2]))
+                            goto FalseExit;
+                    }
+
+                    if (!Number.TryParseHexStrict(buffer[4], buffer[5], ref bytes[1]))
+                        goto FalseExit;
+                    if (!Number.TryParseHexStrict(buffer[6], buffer[7], ref bytes[0]))
+                        goto FalseExit;
+
+                    g._a = Unsafe.ReadUnaligned<int>(bytes);
```

Comment:
> This will return incorrect results on big-endian systems

Are you sure? Why then all the checks (CI) were successful?

Reply:
Because of we do not have test coverage for big-endian in the CI as I have explained here: https://github.com/dotnet/corefx/issues/33684#issuecomment-441412695

Reply:
Ouch! It was not necessary to do this when I was sick. I confused further in the discussion of the Big Endian and ARM64 :(
Sorry for that.